lib/GenomeFileUtil/core/GenomeToGFF.py

```diff
@@ -87,224 +77,160 @@ def get_gff_handle(self, data, output_dir):
             return None
 
         print('pulling cached GFF file from Shock: '+str(data['gff_handle_ref']))
-        dfu = DataFileUtil(self.cfg.callbackURL)
-        file_ret = dfu.shock_to_file({'handle_id':data['gff_handle_ref'],
-                                      'file_path':output_dir,
-                                      'unpack': 'unpack'})
-        return {
-            'file_path': file_ret['file_path']
-        }
-
-
-    ###  see logic from: https://github.com/kbase/KBaseRNASeq/blob/e2d69e4137903c68b5a1fedeab57f7900aad7253/lib/biokbase/RNASeq/KBaseRNASeqImpl.py#L443-L562
-    def build_gff_file(self, getGenomeOptions, output_dir, output_filename, is_gtf):
-
-        # first get subdata needed; forget about the metadata
-        #getGenomeOptions['included_fields'] = []
-        #getGenomeOptions['included_feature_fields'] = ['id', 'type', 'location']
-        getGenomeOptions['no_metadata'] = 1
-        if 'included_fields' in getGenomeOptions:
-            del getGenomeOptions['included_fields']
-        if 'included_feature_fields' in getGenomeOptions:
-            del getGenomeOptions['included_feature_fields']
-
-        api = GenomeAnnotationAPI(self.cfg.callbackURL)
-        genome_data = api.get_genome_v1(getGenomeOptions)['genomes'][0]['data']
+        file_ret = self.dfu.shock_to_file(
+            {'handle_id': data['gff_handle_ref'],
+             'file_path': output_dir,
+             'unpack': 'unpack'})
+        return {'file_path': file_ret['file_path']}
+
+    def build_gff_file(self, genome_data, output_dir, output_filename, is_gtf):
+        def feature_sort(feat):
+            order = ('gene', 'mRNA', 'CDS')
+            if feat['type'] not in order:
+                priority = len(order)
+            else:
+                priority = order.index(feat['type'])
+            return self.get_start(self.get_common_location(
+                feat['location'])), priority
+
+        gff_header = ['seqname', 'source', 'type', 'start', 'end', 'score',
+                      'strand', 'frame', 'attribute']
+        for mrna in genome_data.get('mrnas', []):
+            mrna['type'] = 'mRNA'
+            self.child_dict[mrna['id']] = mrna
+        for cds in genome_data.get('cdss', []):
+            cds['type'] = 'CDS'
+            self.child_dict[cds['id']] = cds
 
         # create the file
+        file_ext = ".gtf" if is_gtf else ".gff"
+        out_file_path = os.path.join(output_dir, output_filename + file_ext)
+        print('Creating file: ' + str(out_file_path))
+
+        # sort every feature in the feat_arrays into a dict by contig
+        features_by_contig = defaultdict(list)
+        for feature in genome_data['features'] + genome_data.get(
+                'non_coding_features', []):
+            if 'type' not in feature:
```

Comment:
Note current API treated all "gene"s the same. There was not a distinction between protein encoding and non encoding genes. I am not sure if this changes your logic.

Reply:
Also the current API only looks in the feature list.  Did you implement new API code?